simulator.py

```diff
@@ -618,71 +674,92 @@ def at(self, location, city, duration):
-                mask_efficacy = self.mask_efficacy * h.mask_efficacy
-
-                infectee = None
-                if self.is_infectious:
-                    ratio = self.asymptomatic_infection_ratio  if self.is_asymptomatic else 1.0
-                    p_infection = self.infectiousness * ratio * (1-mask_efficacy) * proximity_factor
-                    x_human = self.rng.random() < p_infection * CONTAGION_KNOB
-
-                    if x_human and h.is_susceptible:
-                        h.infection_timestamp = self.env.timestamp
-                        self.n_infectious_contacts+=1
-                        Event.log_exposed(h, self, self.env.timestamp)
-                        city.tracker.track_infection('human', from_human=self, to_human=h, location=location, timestamp=self.env.timestamp)
-                        # print(f"{self.name} infected {h.name} at {location}")
-                        infectee = h.name
-
-                elif h.is_infectious:
+                mask_efficacy = h.mask_efficacy # anything that got through their mask will likely get through ours
+                # # I am infectious 
+                # if self.is_infectious:
+                #     ratio = self.asymptomatic_infection_ratio  if self.is_asymptomatic else 1.0
+                #     p_infection = self.infectiousness * ratio * (1-mask_efficacy) * proximity_factor
+                #     x_human = self.rng.random() < p_infection * CONTAGION_KNOB
+
+                #     # Infection
+                #     if x_human and h.is_susceptible:
+                #         h.infection_timestamp = self.env.timestamp
+                #         self.n_infectious_contacts+=1
+                #         Event.log_exposed(h, self, self.env.timestamp)
+                #         city.tracker.track_infection('human', from_human=self, to_human=h, location=location, timestamp=self.env.timestamp)
+                #         # print(f"{self.name} infected {h.name} at {location}")
+                #         infectee = h.name
+
+                # Encountered human is infectious
+                if h.is_infectious:
                     ratio = h.asymptomatic_infection_ratio  if h.is_asymptomatic else 1.0
                     p_infection = h.infectiousness * ratio * (1-mask_efficacy) * proximity_factor # &prob_infectious
-                    x_human = self.rng.random() < p_infection * CONTAGION_KNOB
+                    initial_viral_load += p_infection
+                    x_human = self.rng.random() < initial_viral_load * CONTAGION_KNOB
 
+                    # Infection
                     if x_human and self.is_susceptible:
                         self.infection_timestamp = self.env.timestamp
+                        if initial_viral_load > VIRAL_LOAD_NORMALIZATION:
+                            self.initial_viral_load = 1
```

Comment:
How is incremental viral load different than the earlier version in which we had the precomputed viral load? Why is this necessary?

Reply:
This is *INITIAL* viral load, i.e. the amount of virus that self is exposed to. The magnitude of this payload (also called "viral dose") influences the course of the disease, specifically if you're exposed to a larger amount of virus at infection, you are more likely to have severe symptoms and longer recovery (because your immune system is more immediately overwhelmed, and you have a harder time fighting off the disease overall.

The viral load curve is still precomputed based on this initial viral load, it's just precomputed at the time of infection now, instead of in the init of human.

https://github.com/pg2455/covid_p2p_simulation/pull/81
https://www.zmescience.com/science/viral-load-coronavirus-03042020/
https://www.newscientist.com/article/2238819-does-a-high-viral-load-or-infectious-dose-make-covid-19-worse/
https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30237-1/fulltext
https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30232-2/fulltext